File: LeetCode/cpp/1021_Remove_Outermost_Parentheses.cpp
class Solution {
public:
	string removeOuterParentheses(string s) {
		string answer = "";
		int open = 0, close = 0;

		string temp = "";
		for (int i = 0; i < s.length(); i++) {
			if (s[i] == '(') {
				open++;
				temp += s[i];
			}
			else {
				close++;
				temp += s[i];
			}

			if (open == close) {
				if (open != 1 && close != 1) {
					for (int j = 1; j < temp.size() - 1; j++) {
						answer += temp[j];
					}
				}

				temp = "";
				open = 0;
				close = 0;
			}
		}

		return answer;
	}
};

File: LeetCode/java/1021_Remove_Outermost_Parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        StringBuilder answer = new StringBuilder();
        int open = 0,  close = 0;

        StringBuilder temp = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                open++;
                temp.append(s.charAt(i));
            } else {
                close++;
                temp.append(s.charAt(i));
            }

            if (open == close) {
                if (open != 1 && close != 1) {
                    for (int j = 1; j < temp.length() - 1; j++) {
                        answer.append(temp.charAt(j));
                    }
                }

                temp = new StringBuilder();
                open = 0;
                close = 0;
            }
        }

        return answer.toString();
    }
}